payment-failure/src/main/scala/com/gu/identity/paymentfailure/BrazeEmailService.scala

```diff
@@ -90,13 +90,15 @@ class BrazeEmailServiceWithAbTest(
 
   def sendEmail(emailData: IdentityBrazeEmailData): Either[Throwable, BrazeResponse] = {
     logger.info(s"sending email for test ${variantGenerator.abTest}")
-    for {
+    (for {
       variant <- variantGenerator.generateVariant(emailData.externalId, emailData.emailAddress)
       customFields = variantToCustomFields(variant)
       response <- sendEmailWithCustomFields(emailData, customFields)
     } yield {
-      logger.info(s"braze email sent with encrypted email test data - variant data: $variant")
+      logger.info(s"braze email sent with variant data for test - variant data: $variant")
       response
+    }).recoverWith {
+      case _: UserNotInVariantRange => sendEmailWithCustomFields(emailData, customFields = Map.empty)
```

Comment:
If the error returned in the for comprehension is a result of the user not being in the variant range, still send them the email but without the auto sign in token. If the error is not a UserNotInVariantRange error, the email will not be sent.

Reply:
Great! One suggestion: maybe amend the log statement at the start of this method: "attempting to send..." and if we get to this point in the code log an info message e.g. "user not in test range, sending regular email". Note that 'cos you've had `UserNotInVariantRange` extend a `RuntimeException` you can pass it as a `Throwable` to the second argument of `logger.info()`.

Reply:
Another suggestions regarding logging, which was introduced by me (not you), but that I think makes sense to resolve before this test is released: to not include the auto sign-in token in any log entries. At the moment it will be on line 98 via the string interpolation of `$variant`. To address this you could change the `toString` implementation of `Variant` e.g. not include the `metadata` field. Alternatively, you could use the [`Show`](https://typelevel.org/cats/typeclasses/show.html) type class provided by cats.

Reply:
Have made these changes, thank you. Here is a screenshot from the logs for where the variant is logged now. 
![Screen Shot 2019-05-03 at 15 15 09](https://user-images.githubusercontent.com/42539745/57143275-6c4d6e00-6db6-11e9-96dc-652d6bc84638.png)